core/src/main/java/org/bitcoinj/core/Peer.java

```diff
@@ -226,6 +227,7 @@ public Peer(NetworkParameters params, VersionMessage ver, PeerAddress remoteAddr
                 @Nullable AbstractBlockChain chain, int downloadTxDependencyDepth) {
         super(params, remoteAddress);
         this.params = Preconditions.checkNotNull(params);
+        this.setBlockLocatorStrategy(new LinearBlockLocatorStrategy());
```

Comment:
No functional change yet because this strategy uses the existing code for constructing the block locator.